src/internal/m365/collection/drive/collections.go

```diff
@@ -772,14 +879,44 @@ func (c *Collections) PopulateDriveCollections(
 			c.CollectionMap[driveID] = map[string]*Collection{}
 			invalidPrevDelta = true
 
+			// Reset collections and stats counts since we're starting over.
 			c.resetStats()
+
+			stats = &driveEnumerationStats{}
 		}
 
 		for _, item := range page {
 			if el.Failure() != nil {
 				break
 			}
 
+			// Check if we got the max number of containers we're looking for and also
+			// processed items for the final container.
+			if limiter.enabled() {
+				if item.GetFolder() != nil || item.GetPackageEscaped() != nil {
+					// Don't check for containers we've already seen.
+					if _, ok := c.CollectionMap[driveID][ptr.Val(item.GetId())]; !ok {
+						cp, err := c.getCollectionPath(driveID, item)
+						if err != nil {
+							el.AddRecoverable(ctx, clues.Stack(err).
+								WithClues(ctx).
+								Label(fault.LabelForceNoBackupCreation))
+
+							continue
+						}
+
+						if cp.String() != lastFolderPath {
+							if limiter.atLimit(stats) {
+								break
+							}
+
+							lastFolderPath = cp.String()
+							stats.numContainers++
+						}
```

Comment:
This might have problems, since a unique folder can show up with multiple collection paths across in-flight changes and multiple pages.  That could cause the limiter to break early.  I think you're better off comparing the folder item IDs instead of collection paths.